packages/api/internal/handlers/sandboxes_list_metrics.go

```diff
@@ -191,21 +191,17 @@ func (a *APIStore) GetSandboxesMetrics(c *gin.Context, params api.GetSandboxesMe
 	properties := a.posthog.GetPackageToPosthogProperties(&c.Request.Header)
 	a.posthog.CreateAnalyticsTeamEvent(team.ID.String(), "listed running instances with metrics", properties)
 
-	sandboxes, _, err := a.getSandboxes(ctx, team.ID, SandboxesListParams{
-		State:    &[]api.SandboxState{api.Running},
-		Metadata: params.Metadata,
-	}, SandboxListPaginationParams{
-		Limit:     nil,
-		NextToken: nil,
-	})
-
+	metadataFilter, err := parseMetadata(params.Metadata)
 	if err != nil {
-		zap.L().Error("Error fetching sandboxes", zap.Error(err))
-		a.sendAPIStoreError(c, http.StatusInternalServerError, fmt.Sprintf("Error returning sandboxes for team '%s': %s", team.ID, err))
+		zap.L().Error("Error parsing metadata", zap.Error(err))
+		a.sendAPIStoreError(c, http.StatusBadRequest, fmt.Sprintf("Error parsing metadata: %s", err))
 
 		return
 	}
 
+	// Get relevant running sandboxes
+	sandboxes := a.getRunningSandboxes(ctx, team.ID, metadataFilter)
```

Comment:
I was bit confused by this, in the code there's now `getSandboxes`, `getRunningSandboxes` and `a.orchestrator.GetSandboxes`

Reply:
I didn't add none of these, they've been there already, we can get rid of getSandboxes though..

Reply:
Removed `getSandboxes`